scripts/benchmark/checks/erc20_interact.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import pathlib
+from typing import (
+    Tuple
+)
+
+from web3 import (
+    Web3
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+    to_int,
+)
+
+from evm.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from evm.chains.base import (
+    Chain,
+)
+from evm.rlp.blocks import (
+    BaseBlock,
+)
+
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+    SECOND_ADDRESS,
+    SECOND_ADDRESS_PRIVATE_KEY
+)
+from utils.compile import (
+    get_compiled_contract
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+FIRST_TX_GAS_LIMIT = 1400000
+SECOND_TX_GAS_LIMIT = 100000
+
+CONTRACT_FILE = 'scripts/benchmark/contract_data/erc20.sol'
+CONTRACT_NAME = 'SimpleToken'
+
+W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
+
+
+class Erc20Interact(BaseBenchmark):
+
+    def __init__(self, interact: str, num_blocks: int = 100, num_tx: int = 2) -> None:
+        super().__init__()
+
+        self.num_blocks = num_blocks
+        self.num_tx = num_tx
+        self.interact = interact
+        self.contract_interface = get_compiled_contract(
+            pathlib.Path(CONTRACT_FILE),
+            CONTRACT_NAME
+        )
+
+        self.w3 = Web3()
+        self.addr1 = Web3.toChecksumAddress(FUNDED_ADDRESS)
+        self.addr2 = Web3.toChecksumAddress(SECOND_ADDRESS)
+
+    @property
+    def name(self) -> str:
+        return 'ERC20 ' + self.interact
+
+    def execute(self) -> DefaultStat:
+        total_stat = DefaultStat()
+
+        for chain in get_all_chains():
+
+            # deploy and mine the simple token contract
+            self.simple_token = self.deploy_simple_token(chain)
+            if (self.interact == "transferFrom"):
```

Comment:
Magic strings scare me because the type system can't help you if you just mistype. Let's use an `enum` for the different modes here.

```python
from enum import Enum, auto

class ERC20BenchmarkMode(Enum):

    DeployOnly = auto()
    Transfer = auto()
    SomeOtherMode = auto()
```

Then you can check `self.mode == ERC20BenchmarkMode.DeployOnly` etc.